src/nvim/main.c

```diff
@@ -1623,15 +1623,37 @@ static void read_stdin(void)
   swap_exists_action = SEA_DIALOG;
   no_wait_return = true;
   bool save_msg_didany = msg_didany;
+  buf_T *prev_buf = NULL;
+
+  if (curbuf->b_ffname) {
+    // curbuf is already opened for a file, create a new buffer for stdin #35269
+    buf_T *newbuf = buflist_new(NULL, NULL, 0, 0);
+    if (newbuf == NULL) {
+      ELOG("Unable to allocate buffer for stdin");
```

Comment:
based on other parts of main.c it looks like we can use `semsg` here, which helps avoids confused users. though this should "never" happen in practice.
```suggestion
      semsg("Failed to create buffer for stdin");
```